agent/agent.go

```diff
@@ -206,10 +217,48 @@ func (a *Agent) run(ctx context.Context) {
 			log.G(ctx).Debugf("agent: rebuild session")
 
 			// select a session registration delay from backoff range.
-			delay := time.Duration(rand.Int63n(int64(backoff)))
-			session = newSession(ctx, a, delay)
+			delay := time.Duration(0)
+			if backoff > 0 {
+				delay = time.Duration(rand.Int63n(int64(backoff)))
+			}
+			session = newSession(ctx, a, delay, nodeDescription)
 			registered = session.registered
 			sessionq = a.sessionq
+		case <-nodeUpdateTicker.C:
```

Comment:
I'm not super familiar with this code base, but I'm wondering if this code should be in its own goroutine, simply sending a node description whenever it detects a change.

This `case<-` here would just jump to the session bouncing.

@stevvooe help?